src/NetworkProvider.js

```diff
@@ -2,20 +2,51 @@ import PropTypes from 'prop-types';
 import React, { createContext, useEffect, useState } from 'react';
 import NetInfo from '@react-native-community/netinfo';
 
-export const NetworkContext = createContext({ isConnected: false });
+import appJson from '../app.json';
+import { secrets } from './config';
+
+const defaultIsConnected = false;
+const defaultIsMainserverUp = false;
+
+export const NetworkContext = createContext({
+  isConnected: defaultIsConnected,
+  isMainserverUp: defaultIsMainserverUp
+});
+
+const namespace = appJson.expo.slug;
+
+// https://github.com/react-native-community/react-native-netinfo#netinfoconfiguration
+NetInfo.configure({
+  reachabilityUrl: `${secrets[namespace].serverUrl}/generate_204`,
+  reachabilityTest: async (response) => response.status === 204,
+  reachabilityLongTimeout: 60 * 1000, // 60s
+  reachabilityShortTimeout: 15 * 1000, // 15s
+  reachabilityRequestTimeout: 10 * 1000 // 10s
+});
 
 export const NetworkProvider = ({ children }) => {
-  const [connected, setConnected] = useState(false);
+  const [connected, setConnected] = useState(defaultIsConnected);
+  const [internetReachable, setInternetReachable] = useState(defaultIsMainserverUp);
 
   useEffect(() => {
-    const unsubscribe = NetInfo.addEventListener((state) => setConnected(state.isConnected));
+    // https://github.com/react-native-community/react-native-netinfo#netinfostate
+    const unsubscribe = NetInfo.addEventListener((state) => {
+      setConnected(state.isConnected);
+      setInternetReachable(state.isInternetReachable);
+    });
 
     // returned function will be called on component unmount
     return () => unsubscribe();
   }, []);
 
+  // TODO: remove logs
+  // console.log('NetworkProvider isConnected', connected);
+  // console.log('NetworkProvider isMainserverUp', internetReachable);
+
   return (
-    <NetworkContext.Provider value={{ isConnected: connected }}>{children}</NetworkContext.Provider>
+    <NetworkContext.Provider value={{ isConnected: connected, isMainserverUp: internetReachable }}>
+      {children}
+    </NetworkContext.Provider>
   );
 };
 
```

Comment:
this is something that takes some time to explain, would you be willing to go line by line with me ? I can read about it here mean while https://github.com/react-native-community/react-native-netinfo#readme

Reply:
ok, we will try that.

Reply:
this is not necessary anymore. Read the article is enough for the moment , thanks for you explanations